File: backend/routers/v1/tool.go
package v1

import (
	toolAPI "devops-platform/internal/modules/tool/api"
	"devops-platform/internal/middleware"

	"github.com/gin-gonic/gin"
)

func registerToolRoutes(r *gin.RouterGroup) {
	toolAPI.InstallToolRoutes(r)

	g := r.Group("/tools")
	queryPermission := middleware.RequirePermission("tool", "list")
	updatePermission := middleware.RequirePermission("tool", "update")

	// Templates
	g.GET("/templates", queryPermission, toolAPI.ListTemplates)
	g.GET("/templates/:id", queryPermission, toolAPI.GetTemplate)
	g.POST("/templates", updatePermission,
		middleware.SetAuditOperation("创建工具模板"),
		toolAPI.SaveTemplate)
	g.PUT("/templates/:id", updatePermission, toolAPI.SaveTemplate)
	g.DELETE("/templates/:id", updatePermission,
		middleware.SetAuditOperation("删除工具模板"),
		toolAPI.DeleteTemplate)

	// Template Versions
	g.GET("/templates/:id/versions", queryPermission, toolAPI.ListTemplateVersions)
	g.POST("/templates/:id/versions", updatePermission,
		middleware.SetAuditOperation("创建模板版本"),
		toolAPI.SaveTemplateVersion)
	g.DELETE("/templates/:id/versions/:versionId", updatePermission,
		middleware.SetAuditOperation("删除模板版本"),
		toolAPI.DeleteTemplateVersion)
}

File: frontend/src/api/tool.js
import request from './request'

export const getToolList = (category) => request.get('/tools', { params: { category } })
export const getToolDetail = (id) => request.get(`/tools/${id}`)
export const installTool = (id, data) => request.post(`/tools/${id}/install`, data)
export const checkToolStatus = (id, data) => request.post(`/tools/${id}/check`, data)
export const getInstallations = (hostId) => request.get('/tools/installations', { params: { hostId } })

// Templates
export const listTemplates = (params) => request.get('/tools/templates', { params })
export const getTemplate = (id) => request.get(`/tools/templates/${id}`)
export const saveTemplate = (data) => request.post('/tools/templates', data)
export const updateTemplate = (id, data) => request.put(`/tools/templates/${id}`, data)
export const deleteTemplate = (id) => request.delete(`/tools/templates/${id}`)

// Versions
export const listTemplateVersions = (id) => request.get(`/tools/templates/${id}/versions`)
export const saveTemplateVersion = (id, data) => request.post(`/tools/templates/${id}/versions`, data)
export const deleteTemplateVersion = (templateId, versionId) => request.delete(`/tools/templates/${templateId}/versions/${versionId}`)
